roofit/roofitcore/inc/RooFuncWrapper.h

```diff
@@ -0,0 +1,79 @@
+/*
+ * Project: RooFit
+ * Authors:
+ *   Garima Singh, CERN 2022
+ *
+ * Copyright (c) 2022, CERN
+ *
+ * Redistribution and use in source and binary forms,
+ * with or without modification, are permitted according to the terms
+ * listed in LICENSE (http://roofit.sourceforge.net/license.txt)
+ */
+
+#ifndef RooFit_RooFuncWrapper_h
+#define RooFit_RooFuncWrapper_h
+
+#include "TROOT.h"
+#include "TSystem.h"
+#include "RooAbsReal.h"
+#include "RooGlobalFunc.h"
+#include "RooMsgService.h"
+#include "RooRealVar.h"
+#include "RooListProxy.h"
+
+#include <memory>
+#include <string>
+
+/// @brief  A wrapper class to store a C++ function of type 'double (*)(double* )'.
+/// @tparam Func Function pointer to the generated function.
+template <typename Func = double (*)(double *)>
+class RooFuncWrapper final : public RooAbsReal {
+public:
+   RooFuncWrapper(std::string funcName, std::string funcBody, RooArgSet const &paramSet)
```

Comment:
```suggestion
   RooFuncWrapper(const char* name, const char* title, std::string const& funcBody, RooArgSet const &paramSet)
```
The convention in RooFit that you need to follow is that the first two constructor arguments are always `const char* name, title`. These need to be forwarded to the RooAbsReal constructor. This makes also the `funcName` argument (and data member) redundant: you can just reuse the RooAbsReal object name (that you can retrieve with `GetName()` later. Also, if you take a `std::string`, it should be `const&`.

Reply:
For the function name, the problem is that it needs to be the same for the funcBody string input. If we use ```GetName()```, then all funcBodies should also be named with that name. Alternatively, we could also ONLY take the body of the function without the signature and build the signature on our own. For example, the object c'tor then looks like so:

*Before*
```cpp
// user forced to use same name in function.
std::string func = "double myGauss(double* x) { return x[0] + x[1]; }";
gaussFunc("myGauss", "myGauss", func, {x, mu, sigma})
```

*After*
```cpp
std::string func = "return x[0] + x[1];";
gaussFunc("myGauss", "myGauss", func, {x, mu, sigma})
```

Reply:
I have added the suggested change in the latest commit just to save time if we use this approach.